src/strategyHelpers/deploymentHelper.ts

```diff
@@ -150,22 +150,37 @@ export async function annotateAndLabelResources(
    resourceTypes: Resource[],
    allPods: any
 ) {
+   const defaultWorkflowFileName = 'k8s-deploy-failed-workflow-annotation'
    const githubToken = core.getInput('token')
-   const workflowFilePath = await getWorkflowFilePath(githubToken)
+   let workflowFilePath
+   try {
+      workflowFilePath = await getWorkflowFilePath(githubToken)
+   } catch (ex) {
+      core.warning(`Failed to extract workflow file name: ${ex}`)
+      workflowFilePath = defaultWorkflowFileName
+   }
 
    const deploymentConfig = await getDeploymentConfig()
    const annotationKeyLabel = getWorkflowAnnotationKeyLabel()
+   try {
+      await annotateResources(
+         files,
+         kubectl,
+         resourceTypes,
+         allPods,
+         annotationKeyLabel,
+         workflowFilePath,
+         deploymentConfig
+      )
+   } catch (ex) {
+      core.warning(`Failed to annotate resources: ${ex} `)
+   }
 
-   await annotateResources(
-      files,
-      kubectl,
-      resourceTypes,
-      allPods,
-      annotationKeyLabel,
-      workflowFilePath,
-      deploymentConfig
-   )
-   await labelResources(files, kubectl, annotationKeyLabel)
+   try {
+      await labelResources(files, kubectl, annotationKeyLabel)
```

Comment:
this is a small thing but it's a lot cleaner to do this:

```TypeScript
await labelResources(files, kubectl, annotationKeyLabel).catch(err => core.warning(`Failed to label resources: ${err}`))
```

Cuts down from 5 lines to 1 and conveys the intent a lot better that this is "optional".

Reply:
good point!